frontend/src/layouts/AppLayout/AppLayout.tsx

```diff
@@ -87,12 +87,26 @@ export const AppLayout = ({ children }: LayoutProps) => {
   const [workspaceMapping, setWorkspaceMapping] = useState<Map<string, string>[]>([]);
   const [workspaceSelected, setWorkspaceSelected] = useState('∞');
   const [totalOnboardingActionsDone, setTotalOnboardingActionsDone] = useState(0);
+  const [hasOrganizations, setHasOrganization] = useState(true);
 
   const { t } = useTranslation();
 
   // TODO(akhilmhdh): This entire logic will be rechecked and will try to avoid
   // Placing the localstorage as much as possible
   // Wait till tony integrates the azure and its launched
+
+
+  // small check. if no organization then the add-project from the sidebar should disappear from noOrganization page.
+  useEffect(() => {
```

Comment:
I think you may be able to turn this into a hook — this way we can avoid `useEffect` altogether.